src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2772,60 +2772,40 @@ void os::jvm_path(char *buf, jint buflen) {
     return;
   }
 
-  if (Arguments::sun_java_launcher_is_altjvm()) {
-    // Support for the java launcher's '-XXaltjvm=<path>' option. Typical
-    // value for buf is "<JAVA_HOME>/jre/lib/<vmtype>/libjvm.so".
-    // If "/jre/lib/" appears at the right place in the string, then
-    // assume we are installed in a JDK and we're done. Otherwise, check
-    // for a JAVA_HOME environment variable and fix up the path so it
-    // looks like libjvm.so is installed there (append a fake suffix
-    // hotspot/libjvm.so).
-    const char *p = buf + strlen(buf) - 1;
-    for (int count = 0; p > buf && count < 5; ++count) {
-      for (--p; p > buf && *p != '/'; --p)
-        /* empty */ ;
+  // Look for JAVA_HOME in the environment.
+  char* java_home_var = ::getenv("JAVA_HOME");
+  if (java_home_var != nullptr && java_home_var[0] != 0) {
+    char* jrelib_p;
```

Comment:
Please remove jre from the name - there is no need to mention jre at all here.

Reply:
Fixed.
Also removed the `-Dsun.java.launcher.is_altjvm=true` option from `gtestMain.cpp`.